test/functional/tests/memory/test_memory_consumption_full_cache.py

```diff
@@ -0,0 +1,106 @@
+# SPDX-License-Identifier: BSD-3-Clause-Clear
+#
+
+import pytest
+
+from api.cas import casadm
+from api.cas.cache_config import CacheLineSize, CacheMode
+from core.test_run import TestRun
+from storage_devices.disk import DiskType, DiskTypeSet, DiskTypeLowerThan
+from test_tools.fio.fio import Fio
+from test_tools.fio.fio_param import ReadWrite, IoEngine
+from test_utils.os_utils import (get_mem_available_from_meminfo,
+                                 get_vmalloc_cas_cache_used_from_vmallocinfo,
+                                 is_memory_within_limit,
+                                 validate_memory_consumption,
+                                 get_metadata_max_size)
+from test_utils.os_utils import sync
+from test_utils.size import Size, Unit
+
+cores_count = 16
+cache_mode_wb = CacheMode.WB
+percentage_tolerance = 5
+
+
+@pytest.mark.parametrizex("cache_line_size", CacheLineSize)
+@pytest.mark.require_disk("cache", DiskTypeSet([DiskType.optane, DiskType.nand]))
+@pytest.mark.require_disk("core", DiskTypeLowerThan("cache"))
+def test_memory_consumption_full_cache(cache_line_size):
+    """
+        title: Test for CAS RAM consumption for metadata for cache usage 100%.
+        description: |
+          Validate CAS RAM consumption for metadata depends on start cache line size
+          for cache usage 100% (with 5% tolerance).
+        pass_criteria:
+          - Successful CAS creation and cores addition.
+          - Filling all cache space successfully.
+          - Memory consumption within limits.
+    """
+    with TestRun.step("Prepare drives and partitions for core devices."):
+        cache_disk = TestRun.disks['cache']
+        cache_disk.create_partitions([Size(50, Unit.GibiByte)])
+        cache_dev = cache_disk.partitions[0]
+
+        core_disk = TestRun.disks['core']
+        parts = [Size(3200, Unit.MebiByte)] * cores_count
+        core_disk.create_partitions(parts)
+        core_devices = core_disk.partitions
+
+    with TestRun.step("Start cache with configuration and add all core devices."):
+        free_mem_before_start = get_mem_available_from_meminfo()
+
+        cache = casadm.start_cache(cache_dev, cache_mode_wb, cache_line_size, force=True)
+
+        core_list = []
+        for dev in core_devices:
+            core_list.append(cache.add_core(dev))
+
+    with TestRun.step("Fill all cache space - run Fio instance on each of cores"):
+        sync()
+        for dev in core_list:
+            (Fio().create_command()
+             .direct()
+             .io_engine(IoEngine.libaio)
+             .size(cache.cache_device.size)
+             .read_write(ReadWrite.read)
+             .target(f"{dev.system_path}")
+             .io_depth(64)
+             .block_size(Size(1, Unit.Blocks4096))
+             ).run()
+
+        occupancy = cache.get_statistics(percentage_val=True).usage_stats.occupancy
+        TestRun.LOGGER.info(f"Cache occupancy: {occupancy}")
+        if occupancy < (100 - percentage_tolerance):
```

Comment:
Just FYI - in `math` module there's a method named `isclose`, which is fine to use with calculations with tolerance. However, it may not be the best in this particular case.